api/src/api/package.rs

```diff
@@ -1018,14 +1020,26 @@ document.addEventListener("click", (e) => {
 });
   "#;
 
-  Ok(ApiPackageVersionDocs {
-    css: Cow::Borrowed(deno_doc::html::STYLESHEET),
-    script: Cow::Borrowed(FIXED_SCRIPT),
-    breadcrumbs: docs.breadcrumbs,
-    sidepanel: docs.sidepanel,
-    main: docs.main,
-    version: ApiPackageVersion::from(version),
-  })
+      Ok(respond_json(
+        &ApiPackageVersionDocs {
+          css: Cow::Borrowed(deno_doc::html::STYLESHEET),
+          script: Cow::Borrowed(FIXED_SCRIPT),
+          breadcrumbs: docs.breadcrumbs,
+          sidepanel: docs.sidepanel,
+          main: docs.main,
+          version: ApiPackageVersion::from(version),
+        },
+        StatusCode::OK,
+      ))
+    }
+    GeneratedDocsOutput::Redirect(href) => Ok(
+      Response::builder()
```

Comment:
Rather than returning an actual redirect here, maybe return a JSON of the format `{ "redirect": "<the new url>" }`? Would simplify the handling logic in the frontend significantly.